Formula/coder.rb

```diff
@@ -0,0 +1,19 @@
+class Coder < Formula
+  desc "Tool for provisioning self-hosted development environments with Terraform"
+  homepage "https://coder.com"
+  url "https://github.com/coder/coder/archive/refs/tags/v2.0.2.tar.gz"
+  sha256 "a9c2a996cb916ef0d74c1cdd2ccf30f95b9953618ce2d504a7477eb2b8b655c2"
+  license "AGPL-3.0-only"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args(ldflags: "-s -w"), "./cmd/coder"
```

Comment:
Could this formula download the release binary from GitHub instead? IDK what homebrew's policy is on downloading binaries instead of building them manually. Building it this way means the binary is a "slim" binary and doesn't have the `server` subcommand which a lot of users will want to use.

In v1 we did this: https://github.com/coder/homebrew-coder/blob/cd200308e822e053f181ad281856c990428d3607/coder-cli.rb

If it can't download the binaries, then it should probably call `make gen/mark-fresh && make "build/coder_$(./scripts/version.sh)_darwin_amd64"` (or `arm64`) instead which requires more tools installed like node.js etc.

Reply:
formulas in homebrew-core must build from source. they'll get bottled, and delivered as prebuilt binaries to end-users, but the formula itself can't be written that way.